packages/ember-data/lib/system/model/model.js

```diff
@@ -901,4 +902,19 @@ Model.reopenClass({
   modelName: null
 });
 
+if (Ember.platform.hasPropertyAccessors) {
+  var errorDeprecationShown = false;
+
+  Ember.defineProperty(Model.prototype, 'error', {
+    enumerable: false,
+    get: function() {
+      if (!errorDeprecationShown) {
+        Ember.deprecate('DS.Model#error has been deprecated please use adapterError instead');
+        errorDeprecationShown = true;
+      }
+      return Ember.get(this, 'adapterError');
+    }
+  });
+}
```

Comment:
If someone defines an error attribute I don't think we should log the deprecation because presumably they want to use the error property as an attribute and are not accidentally trying to reference `adapterError`.